festim/hydrogen_transport_problem.py

```diff
@@ -579,12 +579,30 @@ def create_formulation(self):
                 * self.dx(source.volume.id)
             )
 
-            # add fluxes
-            # TODO implement this
-            # for bc in self.boundary_conditions:
-            #     pass
-            #     if bc.species == spe and bc.type != "dirichlet":
-            #         formulation += bc * v * self.ds
+        # add fluxes
+        # TODO implement this
+        # for bc in self.boundary_conditions:
+        #     pass
+        #     if bc.species == spe and bc.type != "dirichlet":
+        #         formulation += bc * v * self.ds
+
+        # check if each species is defined in all volumes
+        if not self.settings.transient:
+            for spe in self.species:
+                # if species mobile, already defined in diffusion term
+                if not spe.mobile:
+                    not_defined_in_volume = self.volume_subdomains.copy()
+                    for vol in self.volume_subdomains:
+                        # check reactions
+                        for reaction in self.reactions:
+                            if vol == reaction.volume:
+                                not_defined_in_volume.remove(vol)
```

Comment:
What happened to sources?

Do we have a test for this case:

- 2 subdomains
- 2 species (one mobile, one immobile)
- 1 reaction in subdomain 1 only (involving the two species)
-  1 source term in subdomain 2 for the immobile species only

I think this should be allowed. In the current implementation, you would apply c_2 = 0 in subdomain 2 even though you have a source term there!

Reply:
Oh thinking about this more, since this is only for steady state, I think that this case would be ill-posed anyway. Let's keep it like this!